src/Voron/Data/CompactTrees/CompactTree.Dictionaries.Random.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using Sparrow;
+
+namespace Voron.Data.CompactTrees;
+
+
+
+unsafe partial class CompactTree
```

Comment:
Why is it a partial? Cannot `RandomKeyScanner` be defined on its own? Also making the file name consistent with the class / struct name inside simplifies the search

Reply:
The reason they are internal to `CompactTree` is for consistency. Some of them use internal tools which are not supposed to be exposed like `IteratorCursorState`. That we can customize how we intend to run the process (random, branch or sequential) doesnt mean that the user should be implement them outside of the `CompactTree` itself (at least for now, the API to interact is very crude).

Reply:
Ok, I see. What do you think about naming the file `CompactTree.Dictionaries.RandomKeyScanner.cs` since it contains `RandomKeyScanner` struct? Right now if I want to quickly check its implementation on github I need to search for a file which name doesn't contain "RandomKeyScanner". (I mean this file and other similar ones as well).

Reply:
Ok, now I understand the rationale of why you want it that way. I used a more semantic meaning for the naming. You want a more search friendly but without semantics (by product of been consistent). I like more the semantic approach (because I used github search usually), but if we are applying the search friendly over the codebase consistency matters. 

Pushed a consistent version in the latest commit.

Reply:
Cool. Thanks